### PR TITLE
Add missing includes

### PR DESCRIPTION
**Pull request by ashtum**

### CODE REVIEW

File: include/boost/property_tree/json_parser/detail/parser.hpp [hunk 1]
```diff
@@ -5,6 +5,7 @@
 
 #include <boost/core/ref.hpp>
 #include <boost/bind/bind.hpp>
+#include <boost/bind/placeholders.hpp>
```
Comment (ashtum) [inline include/boost/property_tree/json_parser/detail/parser.hpp:8]: @pdimov, I've initially added this header to address https://github.com/boostorg/property_tree/pull/101, but it looked incorrect on second thought. I believe it should be the user's responsibility to manage the definition of `BOOST_BIND_NO_PLACEHOLDERS` and it has nothing to with including `boost/bind/placeholders.hpp`. what do you think?

**Comment by pdimov** [comment]
It's hard to say what's the history here.

I suppose this dates from the days when including `<boost/bind.hpp>` made `_1` et al available in the global namespace, which later conflicted with `using namespace std::placeholders;` which also made `_1` available.

So I assume people who were defining `BOOST_BIND_NO_PLACEHOLDERS` did that in order to be able to use `using namespace std::placeholders;` in their code. But doing so broke `parser.hpp`.

This problem will not occur today, because `<boost/bind/bind.hpp>` leaves the placeholders defined in `namespace boost::placeholders`, avoiding the conflict. But for backward compatibility, we may want to still support the configuration with `BOOST_BIND_NO_PLACEHOLDERS` defined.

So adding this include is probably correct, but we need a test that shows why.

**Comment by pdimov** [comment]
As a general rule, every change that does not immediately fix a test failure needs to follow the procedure of: first commit adds a test that fails without the change, second commit adds the change that fixes the test failure.

File: test/Jamfile.v2 [hunk 2]
```diff
@@ -20,6 +20,7 @@ run test_rapidxml.cpp ;
 run test_info_parser.cpp ;
 run test_json_parser.cpp ;
 run test_json_parser2.cpp ;
+run test_json_parser3.cpp ;
```
Comment (pdimov) [inline test/Jamfile.v2:23]: This should probably be `compile` instead of `run`, as we aren't testing anything besides compilation.

**Comment by pdimov** [comment]
Would you please be able to extract the new test into a separate PR? I'd like to merge the original contribution (#101) as a form of giving credit on top of it.

**Pull request merged**